### PR TITLE
Make most textbox carets movable

### LINKED ISSUE #6484: Some TextBoxes should be able to move the caret between letters
**Describe the bug:** TextBoxes like the settings' one, the _filter_ one from the music controller and the social/osu!direct's one can't move the caret between letters. I think they should be able to do that, like in the chat overlay (except for the carousel's one, because with <kbd>left</kbd> and <kbd>right</kbd> you change beatmap): it could be useful for typos (especially when searching for payers/maps in the social/osu!direct's one, instead of deleting the entire text with <kbd>Escape</kbd>).

**osu!lazer version:** 2019.1011.0

### PR DESCRIPTION
**Pull request by Joehuu**
Resolves https://github.com/ppy/osu/issues/6484.

### CODE REVIEW

**Review by frenzibyte — DISMISSED** [review]

**Comment by bdach** [comment]
Just for completeness' sake, here's what I think is an exhaustive list of the locations whose textboxes will now have movable carets (compiled purely by looking through references):

* `osu.Game.Overlays.SettingsPanel`
* `osu.Game.Overlays.Chat.Selection.ChannelSelectionOverlay`
* All users of `osu.Game.Overlays.Music.FilterControl`:
	- `osu.Game.Overlays.Music.PlaylistOverlay`
* All users of `osu.Game.Overlays.SearchableList.SearchableListFilterControl` and derivatives:
	- `osu.Game.Overlays.DirectOverlay`
	- `osu.Game.Overlays.SocialOverlay`
	- `osu.Game.Screens.Multi.Lounge.LoungeSubScreen`

**Comment by smoogipoo** [comment]
Should be disabled in `SettingsPanel` due to `SliderBars` being adjustable via left-right.

**Comment by bdach** [comment]
Also, on a general note, how about making `HandleLeftRightArrows` explicitly settable (be it framework- or lazer-side)? Having to subclass to change behaviour for this one property alone seems suboptimal to me.

**Comment by peppy** [comment]
Almost makes me wonder if we could somehow have a non-focused input path for such cases so we wouldn't have to specify it in the first place (not sure how feasible this is though).

- textbox focused OnKeyDown handling
- rest of heirarchy until textbox
- textbox unfocused OnKeyDown handling

File: osu.Game/Graphics/UserInterface/FixedSearchTextBox.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class FixedSearchTextBox : SearchTextBox
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/FixedSearchTextBox.cs]: What does "fixed" mean here?

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Graphics/UserInterface/FixedSearchTextBox.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Graphics/UserInterface/FixedSearchTextBox.cs]: The caret is "fixed". Sounds misleading now, but wasn't too sure of what to name it in the first place.

Comment (peppy) [reply osu.Game/Graphics/UserInterface/FixedSearchTextBox.cs]: `SeekLimitedSearchTextBox` maybe... hard to come up with a good name

**Review by peppy — APPROVED** [review]

**Pull request merged**